harvester/DBInterface.py

```diff
@@ -1,499 +1,600 @@
 						abort_after_numerrors=?,max_records_updated_per_run=?,update_log_after_numitems=?,record_refresh_days=?,repo_refresh_days=?
-						WHERE repository_id=?"""), (repo_url, repo_set, repo_name, repo_type, repo_thumbnail, time.time(), item_url_pattern, enabled, 
-						abort_after_numerrors,max_records_updated_per_run,update_log_after_numitems,record_refresh_days,repo_refresh_days,repo_id) )
-				except self.dblayer.IntegrityError:
-					# record already present in repo
-					return repo_id
-			else:
-				# Create new repo record
-				try:
-					self.logger.debug("This repo did not exist in the database; adding")
-					if self.dbtype == "postgres":
-						cur.execute(self._prep("""INSERT INTO repositories 
+						WHERE repository_id=?"""), (
+                        repo_url, repo_set, repo_name, repo_type, repo_thumbnail, time.time(), item_url_pattern,
+                        enabled,
+                        abort_after_numerrors, max_records_updated_per_run, update_log_after_numitems,
+                        record_refresh_days,
+                        repo_refresh_days, repo_id))
+                except self.dblayer.IntegrityError as e:
+                    # record already present in repo
+                    self.logger.error("Integerity error in update %s " % (e))
+                    return repo_id
+            else:
+                # Create new repo record
+                try:
+                    self.logger.debug("This repo does not exist in the database; adding")
+                    if self.dbtype == "postgres":
+                        cur.execute(self._prep("""INSERT INTO repositories 
 							(repository_url, repository_set, repository_name, repository_type, repository_thumbnail, last_crawl_timestamp, item_url_pattern, enabled,
 							abort_after_numerrors,max_records_updated_per_run,update_log_after_numitems,record_refresh_days,repo_refresh_days) 
-							VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?) RETURNING repository_id"""), (repo_url, repo_set, repo_name, repo_type, repo_thumbnail, time.time(), item_url_pattern, enabled,
-							abort_after_numerrors,max_records_updated_per_run,update_log_after_numitems,record_refresh_days,repo_refresh_days))
-						res = cur.fetchone()
-						repo_id = int(res[0])
-
-					if self.dbtype == "sqlite":
-						cur.execute(self._prep("""INSERT INTO repositories 
+							VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?) RETURNING repository_id"""), (
+                            repo_url, repo_set, repo_name, repo_type, repo_thumbnail, time.time(), item_url_pattern,
+                            enabled,
+                            abort_after_numerrors, max_records_updated_per_run, update_log_after_numitems,
+                            record_refresh_days, repo_refresh_days))
+                        cur.execute("SELECT CURRVAL('repositories_id_sequence')")
```

Comment:
Not totally sure how this works but you have more experience with the Postgres codepaths than I do at this point